docs/resources/administrative_unit.md

```diff
@@ -0,0 +1,51 @@
+---
+subcategory: "Administrative Units"
+---
+
+# Resource: azuread_administrative_unit
+
+Manages an Administrative Unit within Azure Active Directory.
+
+## API Permissions
+
+The following API permissions are required in order to use this resource.
+
+When authenticated with a service principal, this resource requires one of the following application roles: `AdministrativeUnit.ReadWrite.All` or `Directory.ReadWrite.All`
+
+When authenticated with a user principal, this resource requires one of the following directory roles: `Privileged Role Administrator` or `Global Administrator`
+
+## Example Usage
+
+```terraform
+resource "azuread_administrative_unit" "example" {
+  display_name = "Example-AU"
+  description  = "Just an example"
+  visibility   = "Public"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `description` - (Optional) The description of the administrative unit.
+* `display_name` - (Required) The display name of the administrative unit.
+* `members` - (Optional) A set of object IDs of members who should be present in this administrative unit. Supported object types are Users or Groups.
+
+!> **Warning** Do not use the `members` property at the same time as the [azuread_administrative_unit_member](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/administrative_unit_member) resource for the same administrative unit. Doing so will cause a conflict and administrative unit members will be removed.
+
+* `visibility` - (Optional) Whether the administrative unit _and_ its members are hidden or publicly viewable in the directory. Must be one of: `Hiddenmembership` or `Public`. Defaults to `Public`.
```

Comment:
is there just two options here? would it make sense to make this a boolean such as `publicly_visible` ?

Reply:
I went back and forth on this one - right now there are two documented values but the analogous field for Groups has _three_ values and I'm half expecting this one to go the same way. I'll reach out and see if I can find out.

Reply:
It doesn't look like there is any imminent support for more values here, so I'll reduce it to a boolean 👍

Reply:
ahh if its going to be a third (valid value that makes sense to expose as sometimes there are ones that make little sense) then yea it makes sense as a string

Reply:
In this case a third value doesn't look imminent. If this feature does evolve in time we can deprecate, I think a boolean is the right call here.